File: configlibrary/src/main/java/com/shmedo/configlibrary/iot/cmd/parser/DataCenterInfoParser.java
package com.shmedo.configlibrary.iot.cmd.parser;

import com.shmedo.configlibrary.iot.enums.IOTCommandType;
import com.shmedo.configlibrary.iot.interfaces.IOTResultParser;
import com.shmedo.configlibrary.iot.model.DataCenterInfo;

import java.util.HashMap;

/**
 * 创建者:   gonghe <br/>
 * 创建时间:  11/18/20 <br/>
 * 描述：   解析数据中心参数
 */
public class DataCenterInfoParser implements IOTResultParser<DataCenterInfo> {
    @Override
    public DataCenterInfo parse(String result) {
        DataCenterInfo info = new DataCenterInfo();
        try {
            String[] keyValues = result.split("&");
            HashMap<String, String> keyValueMap = new HashMap<>();

            for (String keyValue : keyValues) {
                String[] strs = keyValue.split("=");
                if (strs.length < 2) {
                    keyValueMap.put(strs[0], "");
                } else {
                    keyValueMap.put(strs[0], strs[1]);
                }
            }
            info.setCenterid(keyValueMap.getOrDefault("centerid", ""));
            info.setProtocol(keyValueMap.getOrDefault("protocol", ""));
            info.setDatatype(keyValueMap.getOrDefault("datatype", ""));
            info.setPlattype(keyValueMap.getOrDefault("plattype", ""));
            info.setAddr(keyValueMap.getOrDefault("addr", ""));
            info.setPort(keyValueMap.getOrDefault("port", ""));

            info.setDeviceid(keyValueMap.getOrDefault("deviceid", ""));
            info.setDevicekey(keyValueMap.getOrDefault("devicekey", ""));
            info.setHttpaddr(keyValueMap.getOrDefault("httpaddr", ""));
            info.setHttpport(keyValueMap.getOrDefault("httpport", ""));
            info.setProjid(keyValueMap.getOrDefault("projid", ""));
            info.setRegcode(keyValueMap.getOrDefault("regcode", ""));

            info.setType_code(keyValueMap.getOrDefault("type_code", ""));
            info.setCo_address(keyValueMap.getOrDefault("co_address", ""));
            info.setPassword(keyValueMap.getOrDefault("password", ""));
            info.setTaddress(keyValueMap.getOrDefault("taddress", ""));
            info.setHour_report(keyValueMap.getOrDefault("hour_report", ""));
            info.setData_link(keyValueMap.getOrDefault("data_link", ""));

            return info;
        } catch (Exception ex) {
            ex.printStackTrace();
            return null;
        }
    }

    @Override
    public void validate(String result) {

    }

    @Override
    public IOTCommandType commandType() {
        return IOTCommandType.MD_GET_DATA_CENTER;
    }
}

File: lib_cmd/src/main/java/com/shmedo/lib/cmd/base/iot_cmd/parser/common/DataCenterInfoParser.kt
package com.shmedo.lib.cmd.base.iot_cmd.parser.common

import com.shmedo.lib.cmd.base.iot_cmd.enums.IOTCommandType
import com.shmedo.lib.cmd.base.iot_cmd.interfaces.IOTCommandParser
import com.shmedo.lib.cmd.base.iot_cmd.model.common.DataCenterInfo
import com.shmedo.lib.cmd.base.iot_cmd.parser.IOTParser

/**
 * 创建者：gonghe
 * 创建时间：2024/1/9
 * 描述： TODO
 */
@IOTParser
class DataCenterInfoParser: IOTCommandParser<DataCenterInfo> {

    override fun parseKeyValueMap(keyValueMap: Map<String, String>): DataCenterInfo {
        return DataCenterInfo().apply {
            centerid = keyValueMap.getOrDefault("centerid", centerid.toString())
            protocol = keyValueMap.getOrDefault("protocol", protocol)
            datatype = keyValueMap.getOrDefault("datatype", datatype)
            plattype = keyValueMap.getOrDefault("plattype", plattype)
            addr = keyValueMap.getOrDefault("addr", addr)
            port = keyValueMap.getOrDefault("port", port)
            deviceid = keyValueMap.getOrDefault("deviceid", deviceid)
            devicekey = keyValueMap.getOrDefault("devicekey", devicekey)
            httpaddr = keyValueMap.getOrDefault("httpaddr", httpaddr)
            httpport = keyValueMap.getOrDefault("httpport", httpport)
            projid = keyValueMap.getOrDefault("projid", projid)
            regcode = keyValueMap.getOrDefault("regcode", regcode)
            type_code = keyValueMap.getOrDefault("type_code", type_code)
            co_address = keyValueMap.getOrDefault("co_address", co_address)
            password = keyValueMap.getOrDefault("password", password)
            taddress = keyValueMap.getOrDefault("taddress", taddress)
            hour_report = keyValueMap.getOrDefault("hour_report", hour_report)
            data_link = keyValueMap.getOrDefault("data_link", data_link)
            valid_day = keyValueMap.getOrDefault("valid_day", valid_day)
            reissue_time = keyValueMap.getOrDefault("reissue_time", reissue_time)
        }
    }

    override val commandType: IOTCommandType = IOTCommandType.MD_GET_DATA_CENTER
}